File: app/src/main/java/com/example/krystianwsul/organizator/utils/time/DayOfWeek.java
package com.example.krystianwsul.organizator.utils.time;

import junit.framework.Assert;

import java.text.DateFormatSymbols;
import java.util.Calendar;

public enum DayOfWeek {
    SUNDAY,
    MONDAY,
    TUESDAY,
    WEDNESDAY,
    THURSDAY,
    FRIDAY,
    SATURDAY;

    public static DayOfWeek getDayFromCalendar(Calendar calendar) {
        Assert.assertTrue(calendar != null);

        int day = calendar.get(Calendar.DAY_OF_WEEK);
        return values()[day-1];
    }

    public String toString() {
        return DateFormatSymbols.getInstance().getWeekdays()[this.ordinal() + 1];
    }

    public static DayOfWeek today() {
        return Date.today().getDayOfWeek();
    }
}

File: common/src/commonMain/kotlin/time/DayOfWeek.kt
package com.krystianwsul.common.time

enum class DayOfWeek {
    SUNDAY,
    MONDAY,
    TUESDAY,
    WEDNESDAY,
    THURSDAY,
    FRIDAY,
    SATURDAY;

    companion object {

        private val dateCache = mutableMapOf<Date, DayOfWeek>() // because I can't hold this on a parcelable date

        fun fromDate(date: Date): DayOfWeek {
            if (!dateCache.containsKey(date)) dateCache[date] = values()[date.toDateSoy().dayOfWeekInt]

            return dateCache.getValue(date)
        }

        val set by lazy { values().toSet() }
    }

    override fun toString() = format()
}
